components/common-go/baseserver/server.go

```diff
@@ -0,0 +1,213 @@
+package baseserver
+
+import (
+	"context"
+	"fmt"
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"net"
+	"net/http"
+)
+
+type config struct {
```

Comment:
We should export this so that we can reuse this config in the installer and component config.

We'd want to pull in pprof and prometheus, too.